Automattic/woocommerce-payments
# Add a Last Updated Column to the Currency class

Fixes #2254 

#### Changes proposed in this Pull Request









- Adds a `last_updated` property to the Currency class
- Updates the logic to update this value when currencies are fetched from the cache
- Updates the settings page to read this value and display it on the frontend
- Also fixed a few Psalm violations in the Multi Currency classes.

#### Testing instructions





- Update your sandbox to the latest server code by running `svn up .` from the `~/public_html` folder within your sandbox.
Point your client to your sandbox by disabling API redirection in the WCPay Dev Tools.
- Go to `Settings -> Multi-Currency` and add a few currencies (e.g. EUR, GBP, CAD, VND) to the enabled currencies list using the Add Currencies modal. (If you have already got a few enabled currencies, skip this step).
- Go to `Settings -> General` and change your default store currency to force the cache to be cleared.
- Go back to `Settings -> Multi-Currency` and click 'Edit Currency' on one of the currencies. Check the value of the "last updated" time and verify that it matches the current time (in UTC).
- Wait a few minutes, then go back to `Settings -> General` and change the default store currency again.
- Go back to `Settings -> Multi-Currency` and click 'Edit Currency' on one of the currencies. Check the value of the "last updated" time and verify that it was updated since the last time you checked and that it matches the current time (in UTC).

-------------------
